src/dimension.c

```diff
@@ -1001,24 +1001,32 @@ dimension_interval_to_internal(const char *colname, Oid dimtype, Oid valuetype,
 			interval = get_validated_integer_interval(dimtype, DatumGetInt64(value));
 			break;
 		case INTERVALOID:
-			if (IS_INTEGER_TYPE(dimtype))
+			if (!IS_TIMESTAMP_TYPE(dimtype))
 				ereport(ERROR,
 						(errcode(ERRCODE_INVALID_PARAMETER_VALUE),
-						 errmsg(
-							 "invalid interval: must be an integer type for integer dimensions")));
+						 errmsg("invalid interval type for %s dimension", format_type_be(dimtype)),
+						 errhint("Use an interval of type integer.")));
 
 			interval = interval_to_usec(DatumGetIntervalP(value));
 			break;
 		default:
-			ereport(ERROR,
-					(errcode(ERRCODE_INVALID_PARAMETER_VALUE),
-					 errmsg("invalid interval: must be an interval or integer type")));
+			if (IS_TIMESTAMP_TYPE(dimtype))
+				ereport(ERROR,
+						(errcode(ERRCODE_INVALID_PARAMETER_VALUE),
+						 errmsg("invalid interval type for %s dimension", format_type_be(dimtype)),
+						 errhint("Use an interval of type interval or integer.")));
+			else
+				ereport(ERROR,
+						(errcode(ERRCODE_INVALID_PARAMETER_VALUE),
+						 errmsg("invalid interval type for %s dimension", format_type_be(dimtype)),
+						 errdetail("Use an interval of type integer.")));
```

Comment:
They only differ in the detail message. Probably good to use a `:?` there instead to avoid mistakes when maintaining this code.